pkg/cmd/pr/checks/checks.go

```diff
@@ -0,0 +1,211 @@
+		Remotes:    f.Remotes,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "checks",
+		Short: "Show CI status for a single pull request",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if repoOverride, _ := cmd.Flags().GetString("repo"); repoOverride != "" && len(args) == 0 {
+				return &cmdutil.FlagError{Err: errors.New("argument required when using the --repo flag")}
+			}
+
+			if len(args) > 0 {
+				opts.SelectorArg = args[0]
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return checksRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func checksRun(opts *ChecksOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	pr, _, err := shared.PRFromArgs(apiClient, opts.BaseRepo, opts.Branch, opts.Remotes, opts.SelectorArg)
+	if err != nil {
+		return err
+	}
+
+	if len(pr.Commits.Nodes) == 0 {
+		return nil
+	}
+
+	rollup := pr.Commits.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
+	if len(rollup) == 0 {
+		return nil
+	}
+
+	passing := 0
+	failing := 0
+	pending := 0
+
+	type output struct {
+		mark    string
+		bucket  string
+		name    string
+		elapsed string
+		link    string
+	}
+
+	outputs := []output{}
+
+	for _, c := range pr.Commits.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes {
+		mark := ""
+		bucket := ""
+		state := c.State
+		if state == "" {
+			if c.Status == "COMPLETED" {
+				state = c.Conclusion
+			} else {
+				state = c.Status
+			}
+		}
+		switch state {
+		case "SUCCESS", "NEUTRAL", "SKIPPED":
+			mark = utils.GreenCheck()
+			passing++
+			bucket = "pass"
+		case "ERROR", "FAILURE", "CANCELLED", "TIMED_OUT", "ACTION_REQUIRED":
+			mark = utils.RedX()
+			failing++
+			bucket = "fail"
+		case "EXPECTED", "REQUESTED", "QUEUED", "PENDING", "IN_PROGRESS", "STALE":
+			mark = utils.YellowDash()
+			pending++
+			bucket = "pending"
+		default:
+			panic(fmt.Errorf("unsupported status: %q", state))
+		}
+
+		elapsed := ""
+		zeroTime := time.Time{}
+
+		if c.StartedAt != zeroTime && c.CompletedAt != zeroTime {
+			e := c.CompletedAt.Sub(c.StartedAt)
+			if e > 0 {
+				elapsed = e.String()
+			}
+		}
+
+		link := c.DetailsURL
+		if link == "" {
+			link = c.TargetURL
+		}
+
+		name := c.Name
+		if name == "" {
+			name = c.Context
+		}
+
+		outputs = append(outputs, output{mark, bucket, name, elapsed, link})
+	}
+
+	sort.Slice(outputs, func(i, j int) bool {
+		if outputs[i].bucket == outputs[j].bucket {
+			return outputs[i].name < outputs[j].name
+		} else {
+			if outputs[i].bucket == "fail" {
+				return true
+			} else if outputs[i].bucket == "pending" && outputs[j].bucket == "success" {
+				return true
+			}
+		}
+
+		return false
+	})
+
+	tp := utils.NewTablePrinter(opts.IO)
+
+	for _, o := range outputs {
+		if opts.IO.IsStdoutTTY() {
+			tp.AddField(o.mark, nil, nil)
```

Comment:
Just noticed that this passes in text inclusive of ANSI escape codes into the table printer. This leads to some odd printing on narrow terminals. For the purpose of measuring the width of visible text, right now TablePrinter expects text without any ANSI codes in the 1st argument, but you can pass a colorizer function as the 3rd argument.

Reply:
ah, i forgot long ago what those other parameters are. i may be doing this elsewhere >_> fixed it here